lib/matplotlib/tests/test_scale.py

```diff
@@ -148,3 +150,24 @@ def test_invalid_log_lims():
     with pytest.warns(UserWarning):
         ax.set_ylim(top=-1)
     assert ax.get_ylim() == original_ylim
+
+
+@image_comparison(baseline_images=['function_scales'], remove_text=True,
+                  extensions=['png'], style='mpl20')
+def test_function_scale():
+    def inverse(x):
+        return x**2
+
+    def forward(x):
+        good = x >= 0
+        y = np.full_like(x, np.NaN)
+        y[good] = x[good]**(1/2)
+        return y
```

Comment:
I guess in the same vein something as simple as
```
def forward(x): return x**(1/2)
```
also works (apparently it does in the scales.py example above...)?  In which case you can probably even just pass them as lambdas below...

Reply:
Sure, they can be lambdas, but I don't find that easier to read than just defining the functions....  But main point taken...